src/components/DisplayWeb3.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState, useEffect } from 'react'
+import ReactDOM from 'react-dom'
+
+import { Web3Api } from '../types'
+
+interface DisplayWeb3Props {
+    web3Api: Web3Api
+}
+
+interface ClientVersion {
+    id?: number
+    jsonrpc?: string
+    result: string
+}
+
+export const DisplayWeb3: React.FC<DisplayWeb3Props> = ({ web3Api }) => {
+    const [account, setAccount] = useState<string>()
+
+    const [balance, setBalance] = useState<string>()
+
+    const [networkName, setNetworkName] = useState<string>()
+
+    const [clientVersion, setClientVersion] = useState<ClientVersion>()
+
+    useEffect(() => {
+        const setter = async (): Promise<void> => {
+            const accountsProm = web3Api.getAccounts()
+            const networkNameProm = web3Api.getNetworkName()
+            const clientVersionProm = web3Api.web3.currentProvider.send('web3_clientVersion', [])
+
+            const [account] = await accountsProm
+
+            const [balance, networkName, clientVersion] = await Promise.all([
+                web3Api.getBalance(account),
+                networkNameProm,
+                clientVersionProm,
+            ])
+
+            ReactDOM.unstable_batchedUpdates(() => {
```

Comment:
what is `unstable_batchedUpdates`? 
I'm not super familiar, made me curious the word "unstable"

Reply:
It's what they use internally in event handlers to avoid multiple rerenders.
React-redux also uses it, so even if it's called unstable everyone does it